### PR TITLE
new(Grid): Add horizontal alignment props. Fix offset % definition.

### PR DESCRIPTION
**Pull request by stefhatcher**
to: @milesj @stefhatcher @hayes @alecklandgraf

## Description

these updates help make single column grid layouts more flexible with horizontal alignments:
- `startAlign` for `justify-content: flex-start`
- `centerAlign` for `justify-content: center`
- `endAlign` for `justify-content: flex-end`

also noticed that the offsets were being calculated as `px` instead of `%`. added a fix for that!

updated storybook to demonstrate the changes.

## Motivation and Context

tried to do a single column, centered by offsets and couldn't achieve it. this'll help!

## Testing

storybook examples

## Screenshots

Storybook 2019-04-19 14-20-26
Screen Shot 2019-04-19 at 2 20 55 PM

## Checklist

- My code follows the style guide of this project.
- I have updated or added documentation accordingly.
- I have read the CONTRIBUTING document.

### CODE REVIEW

**Comment by milesj** [comment]
Danger run resulted in 1 fail; to find out more, see the [checks page](https://github.com/airbnb/lunar/runs/106281246).
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 99bdc85117c420550a42b3e91aa176ed84798810
</p>

File: packages/core/src/components/Grid/index.tsx [hunk 1]
```diff
@@ -3,39 +3,62 @@ import { childrenOfType, mutuallyExclusiveTrueProps } from 'airbnb-prop-types';
...
 
 export type Props = {
   /** Vertically align the columns at the bottom. */
   bottomAlign?: boolean;
+  /** Horizontally align the columns to the center. */
+  centered?: boolean;
   /** Columns to render. */
   children: NonNullable<React.ReactNode>;
+  /** Horizontally align the columns to the left (flex-start). */
+  leftAlign?: boolean;
```
Comment (milesj) [inline packages/core/src/components/Grid/index.tsx]: `startAlign`

File: packages/core/src/components/Grid/index.tsx [hunk 2]
```diff
@@ -3,39 +3,62 @@ import { childrenOfType, mutuallyExclusiveTrueProps } from 'airbnb-prop-types';
...
   /** Vertically align the columns in the middle. */
   middleAlign?: boolean;
   /** Reverse the order of columns. */
   reversed?: boolean;
+  /** Horizontally align the columns to the left (flex-end). */
+  rightAlign?: boolean;
```
Comment (milesj) [inline packages/core/src/components/Grid/index.tsx]: `endAlign`

File: packages/core/src/components/Grid/index.tsx [hunk 3]
```diff
@@ -3,39 +3,62 @@ import { childrenOfType, mutuallyExclusiveTrueProps } from 'airbnb-prop-types';
...
 
-const alignProp = mutuallyExclusiveTrueProps('bottomAlign', 'middleAlign', 'topAlign');
+const horizontalAlignProp = mutuallyExclusiveTrueProps('centered', 'leftAlign', 'rightAlign');
+const verticalAlignProp = mutuallyExclusiveTrueProps('bottomAlign', 'middleAlign', 'topAlign');
 
 export type Props = {
   /** Vertically align the columns at the bottom. */
   bottomAlign?: boolean;
+  /** Horizontally align the columns to the center. */
+  centered?: boolean;
```
Comment (milesj) [inline packages/core/src/components/Grid/index.tsx]: `centerAlign`?

File: packages/core/src/components/Grid/Col.tsx [hunk 4]
```diff
@@ -47,7 +47,7 @@ export default withStyles(({ unit }) => {
     };
 
     offsets[`offset${offset}`] = {
-      marginLeft: offset > 0 ? 100 / (12 / offset) : 0,
+      marginLeft: offset > 0 ? `${100 / (12 / offset)}%` : 0,
```
Comment (milesj) [inline packages/core/src/components/Grid/Col.tsx]: Was this actually breaking before?

Comment (stefhatcher) [reply packages/core/src/components/Grid/Col.tsx]: yep! didn't notice it in the examples because of the default `space-between`

Comment (stefhatcher) [reply packages/core/src/components/Grid/Col.tsx]: Screen Shot 2019-04-19 at 6 25 22 PM

Comment (milesj) [reply packages/core/src/components/Grid/Col.tsx]: Nice catch :P

Comment (stefhatcher) [reply packages/core/src/components/Grid/Col.tsx]: 😎

**Comment by stefhatcher** [comment]
weekend bump @milesj

**Pull request merged**